VerseReportingProcessor/MergeCompletedNotificationService.cs

```diff
@@ -0,0 +1,259 @@
+    private readonly IConfiguration _configuration;
+    private readonly ServiceBusProcessor _serviceBusProcessor;
+    private const string Topic = "MergedResult";
+    private const string Subscription = "NotificationHandler";
+    private readonly ActivitySource _activitySource = new(nameof(MergeCompletedNotificationService));
+    private readonly VerseProcessorMetrics _metrics;
+    private readonly OrganizationServiceFactory _serviceFactory;
+    private readonly OptionSetValue RepoActive = new(1);
+    private readonly OptionSetValue RepoPrimary = new(953860000);
+    
+    public MergeCompletedNotificationService(ILogger<MergeCompletedNotificationService> logger, IConfiguration configuration, VerseProcessorMetrics metrics, OrganizationServiceFactory serviceFactory)
+    {
+        _logger = logger;
+        _configuration = configuration;
+        _metrics = metrics;
+        var serviceBusClient = new ServiceBusClient(_configuration["ConnectionStrings:ServiceBus"]);
+        _serviceBusProcessor = serviceBusClient.CreateProcessor(Topic, Subscription);
+        _serviceBusProcessor.ProcessMessageAsync += ProcessMessage;
+        _serviceBusProcessor.ProcessErrorAsync += ProcessError;
+        _serviceFactory = serviceFactory;
+    }
+
+    private Task ProcessError(ProcessErrorEventArgs arg)
+    {
+        _logger.LogError(arg.Exception, "An unhandled exception occurred.");
+        return Task.CompletedTask;
+    }
+
+    private async Task ProcessMessage(ProcessMessageEventArgs arg)
+    {
+        var parentActivityId = arg.Message.ApplicationProperties.TryGetValue("Diagnostic-Id", out var diagnosticId) ? diagnosticId.ToString() : null;
+        using var activity = _activitySource.StartActivity("ProcessMessage", ActivityKind.Consumer, parentActivityId);
+        var body = arg.Message.Body.ToString();
+        var message = JsonSerializer.Deserialize<MergeResult>(body);
+        if (message == null)
+        {
+            _logger.LogError("Invalid message received");
+            return;
+        }
+        _logger.LogInformation("Processing message for a merge for {Language}", message.LanguageCode);
+        var service = await _serviceFactory.GetServiceClientAsync();
+        var messageText = "";
+        var notificationText = "";
+        Guid? newMergedRepoId = null;
+        var tasks = new List<Task>();
+        if (message.Success)
+        {
+            newMergedRepoId = await CreateRepoRecordInPORTForNewlyMerged(service, message);
+            messageText = $"Your merge for {message.LanguageCode} is now complete you can find the result here <a href=\"{message.MergedUrl}\">{message.MergedUrl}</a>";
+            notificationText = $"Your merge for {message.LanguageCode} is now complete.";
+            if (newMergedRepoId != null)
+            {
+                tasks.Add(SwitchPrimaryForRepoInPORT(service, newMergedRepoId.Value, message.MergedRepoPORTIds));
```

Comment:
I don't think we want to change the Primary status of any repos on a merge action.  The combined repo will need a bunch of work and refinement before it's ready to e.g. go up on BIEL.  In fact, we may have multiple rounds of changes as the refinement team works.

I expect changing the Primary status of the source repos and the combined repo will be a manual action, not an automatic one.

Reply:
That was what was in the requirements, now that we have those changed I removed this portion.